front/app/modules/commercial/insights/admin/containers/Reports/Charts/CustomFieldGraph.tsx

```diff
@@ -154,7 +108,32 @@ const CustomFieldsGraph = ({
   intl: { formatMessage },
   className,
 }: Props) => {
-  const [serie, setSerie] = useState<TOutput | null>(null);
+  const { code } = customField.attributes;
+
+  const { data: usersByGenderWithFilters } = useUsersByGender({
+    start_at: startAt,
+    end_at: endAt,
+    project: currentProject,
+    filter_by_participation: true,
+  });
+  const { data: usersByGenderWithoutFilters } = useUsersByGender({});
+
+  const { data: usersByBirthyearWithFilters } = useUsersByBirthyear({
+    start_at: startAt,
+    end_at: endAt,
+    project: currentProject,
+    filter_by_participation: true,
+  });
+  const { data: usersByBirthyearWithoutFilters } = useUsersByBirthyear({});
+
+  const { data: usersByDomicileWithFilters } = useUsersByDomicile({
+    start_at: startAt,
+    end_at: endAt,
+    project: currentProject,
+    filter_by_participation: true,
+  });
+  const { data: usersByDomicileWithoutFilters } = useUsersByDomicile({});
```

Comment:
Obviously if you were to use this component only once, this would be pretty bad- if you only wanted e.g. the birthyear chart by passing this component the `customField` with `code === 'birthyear'`, you would still be fetching all the other domicile and gender data too. In this case I guess it's ok since in the context where it's used, it's only used for all of those fields, so the data will all be cached, but definitely not the cleanest implementation

Reply:
Good point!